### PR TITLE
Remove MSEDataConsistency

### LINKED ISSUE #402: Consider removing MSEDataConsistency
We could either consider removing MSEDataConsistency and replacing all uses with L2NormSquared or changing it to an alias of L2NormSquared with different parameter names.

### PR DESCRIPTION
**Pull request by rkcatarina**
Closes #402

### CODE REVIEW

**Comment by fzimmermann89** [comment]
👍 

I would keep the names as mse, i.e.
`mse = L2NormSquared(....)`

Furthermore, I would change the default of divide_by_n to True. Thinking a bit about it, there are almost no use case where it is not better to divide.

File: examples/t1_mapping_with_grad_acq.py [hunk 1]
```diff
@@ -173,10 +173,10 @@
 
 # %% [markdown]
 # ### Loss function
-# As a loss function for the optimizer, we calculate the mean-squared error between the image data $x$ and our signal
+# As a loss function for the optimizer, we calculate the squared L2 norm between the image data $x$ and our signal
```
Comment (ckolbPTB) [inline examples/t1_mapping_with_grad_acq.py]:
```suggestion
# As a loss function for the optimizer, we calculate the mean squared error between the image data $x$ and our signal
```

File: src/mrpro/operators/functionals/MSE.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+from collections.abc import Sequence
+
+import torch
+
+from mrpro.operators.functionals.L2NormSquared import L2NormSquared
+
+
+class MSE(L2NormSquared):
+    r"""Functional class for the mean square error.
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/MSE.py]:
```suggestion
    r"""Functional class for the mean squared error.
```

File: src/mrpro/operators/functionals/MSE.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    def __init__(
+        self,
+        weight: torch.Tensor | complex = 1.0,
+        target: torch.Tensor | None | complex = None,
+        dim: int | Sequence[int] | None = None,
+        divide_by_n: bool = True,
+        keepdim: bool = False,
+    ) -> None:
+        r"""Initialize a Functional.
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/MSE.py]:
```suggestion
        r"""Initialize MSE functional.
```

File: src/mrpro/operators/functionals/MSE.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        We assume that functionals are given in the form
+        :math:`f(x) = \phi ( weight ( x - target))`
+        for some functional :math:`\phi`.
+
+        Parameters
+        ----------
+         functional
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/MSE.py]: use doc string from here:
https://github.com/PTB-MR/mrpro/blob/798f1e26d4736127fe697319b3354e0b7f703105/src/mrpro/operators/Functional.py#L67

File: src/mrpro/operators/functionals/MSE.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        self,
+        weight: torch.Tensor | complex = 1.0,
+        target: torch.Tensor | None | complex = None,
+        dim: int | Sequence[int] | None = None,
+        divide_by_n: bool = True,
+        keepdim: bool = False,
+    ) -> None:
+        r"""Initialize a Functional.
+
+        We assume that functionals are given in the form
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/MSE.py]: Replace with
The MSE functional is given by 
:math:`f: C^N -> [0, \infty), x -> \| W (x-b)\|_2^2`,
    where :math:`W` is either a scalar or tensor that corresponds to a (block-) diagonal operator
    that is applied to the input. For more details also see :class:`mrpro.operators.functionals.L2NormSquared`

File: src/mrpro/operators/functionals/MSE.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+import torch
+
+from mrpro.operators.functionals.L2NormSquared import L2NormSquared
+
+
+class MSE(L2NormSquared):
+    r"""Functional class for the mean square error.
+
+    This makes use of the functional L2NormSquared.
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/MSE.py]:
```suggestion
    This makes use of the functional :class:`mrpro.operators.functionals.L2NormSquared`.
```

**Review by ckolbPTB — COMMENTED** [review]
In most cases, consider setting divide_by_n to true to be independent of input size. Alternatively the functional :class:`mrpro.operators.functionals.L2NormSquared` can be used.

should also be added to the L2NormSquared functional.

File: src/mrpro/operators/functionals/MSE.py (hunk 6 above)
Comment (fzimmermann89) [inline src/mrpro/operators/functionals/MSE.py]:
```suggestion
```

File: src/mrpro/operators/functionals/MSE.py (hunk 5 above)
Comment (fzimmermann89) [reply src/mrpro/operators/functionals/MSE.py]: 1/N should maybe be mentioned in the eqoation already

File: src/mrpro/operators/functionals/L1Norm.py [hunk 7]
```diff
@@ -13,6 +13,9 @@ class L1Norm(ElementaryProximableFunctional):
     where W is a either a scalar or tensor that corresponds to a (block-) diagonal operator
     that is applied to the input.
 
+    In most cases, consider setting divide_by_n to true to be independent of input size.
+    Alternatively the functional :class:`mrpro.operators.functionals.L2NormSquared` can be used.
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/L1Norm.py]:
```suggestion
```

File: src/mrpro/operators/functionals/L1NormViewAsReal.py [hunk 8]
```diff
@@ -15,6 +15,9 @@ class L1NormViewAsReal(ElementaryProximableFunctional):
     If the parameter `weight` is real-valued, :math:`W_r` and :math:`W_i` are both set to `weight`.
     If it is complex-valued, :math:`W_r` and :math:`W_I` are set to the real and imaginary part, respectively.
 
+    In most cases, consider setting divide_by_n to true to be independent of input size.
+    Alternatively the functional :class:`mrpro.operators.functionals.L2NormSquared` can be used.
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/L1NormViewAsReal.py]:
```suggestion
```

File: src/mrpro/operators/functionals/L2NormSquared.py [hunk 9]
```diff
@@ -15,6 +15,8 @@ class L2NormSquared(ElementaryProximableFunctional):
     reconstruction when using a density-compensation function for k-space pre-conditioning,
     for masking of image data, or for spatially varying regularization weights.
 
+    In most cases, consider setting divide_by_n to true to be independent of input size.
+
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/L2NormSquared.py]:
```suggestion
Alternatively the functional :class:`mrpro.operators.functionals.MSE` can be used.
```

File: src/mrpro/operators/functionals/MSE.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        target: torch.Tensor | None | complex = None,
+        dim: int | Sequence[int] | None = None,
+        divide_by_n: bool = True,
+        keepdim: bool = False,
+    ) -> None:
+        r"""Initialize MSE Functional.
+
+        The MSE functional is given by
+        :math:`f: C^N -> [0, \infty), x -> 1/N \| W (x-b)\|_2^2`,
+        where :math:W is either a scalar or tensor that corresponds to a (block-) diagonal operator
```
Comment (ckolbPTB) [inline src/mrpro/operators/functionals/MSE.py]:
```suggestion
        where :math:`W` is either a scalar or tensor that corresponds to a (block-) diagonal operator
```

**Comment by fzimmermann89** [comment]
pre-commit.ci autofix

File: src/mrpro/operators/functionals/MSE.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+        dim: int | Sequence[int] | None = None,
+        divide_by_n: bool = True,
+        keepdim: bool = False,
+    ) -> None:
+        r"""Initialize MSE Functional.
+
+        The MSE functional is given by
+        :math:`f: C^N -> [0, \infty), x -> 1/N \| W (x-b)\|_2^2`,
+        where :math:`W` is either a scalar or tensor that corresponds to a (block-) diagonal operator
+        that is applied to the input. The squared norm is then divided by N.
```
Comment (fzimmermann89) [inline src/mrpro/operators/functionals/MSE.py]:
```suggestion
        that is applied to the input. The division by `N` can be disabled by setting `divide_by_n=False`
```

**Review by ckolbPTB — APPROVED** [review]

**Pull request merged**